gdal/alg/gdalwarpkernel.cpp

```diff
@@ -6102,6 +6110,64 @@ static void GWKAverageOrModeThread( void* pData)
                         bHasFoundDensity = true;
                     }
                 }  // GRA_Average.
+                // poWK->eResample == GRA_RMS.
+                if( nAlgo == GWKAOM_RMS )
+                {
+                    double dfTotalReal = 0.0;
+                    double dfTotalImag = 0.0;
+                    double dfTotalWeight = 0.0;
+                    double dfR = 0.0;
+                    double dfI = 0.0;
+                    // This code adapted from GDALDownsampleChunk32R_AverageT()
+                    // in gcore/overview.cpp.
+                    for( int iSrcY = iSrcYMin; iSrcY < iSrcYMax; iSrcY++ )
+                    {
+                        const double dfWeightY = COMPUTE_WEIGHT_Y(iSrcY);
+                        iSrcOffset = iSrcXMin + static_cast<GPtrDiff_t>(iSrcY) * nSrcXSize;
+                        for( int iSrcX = iSrcXMin; iSrcX < iSrcXMax; iSrcX++, iSrcOffset++ )
+                        {
+                            if( poWK->panUnifiedSrcValid != nullptr
+                                && !(poWK->panUnifiedSrcValid[iSrcOffset>>5]
+                                     & (0x01 << (iSrcOffset & 0x1f))) )
+                            {
+                                continue;
+                            }
+
+                            if( GWKGetPixelValue(
+                                    poWK, iBand, iSrcOffset,
+                                    &dfBandDensity, &dfValueRealTmp,
+                                    &dfValueImagTmp ) &&
+                                dfBandDensity > BAND_DENSITY_THRESHOLD )
+                            {
+                                const double dfWeight = COMPUTE_WEIGHT(iSrcX, dfWeightY);
+                                dfTotalWeight += dfWeight;
+                                if (bIsComplex)
+                                {
+                                    dfTotalReal += (dfValueRealTmp * dfValueRealTmp - dfValueImagTmp *dfValueImagTmp) * dfWeight;
+                                    dfTotalImag += 2. * dfValueRealTmp * dfValueImagTmp * dfWeight;
+                                }
+                                else
+                                    dfTotalReal += dfValueRealTmp * dfValueRealTmp * dfWeight;
+                            }
+                        }
+                    }
+
+                    if( dfTotalWeight > 0 )
+                    {
+                        if (bIsComplex)
+                        {
+                            dfR = dfTotalReal / dfTotalWeight;
+                            dfI = dfTotalImag / dfTotalWeight;
+                            dfValueReal = sqrt((dfR + sqrt(dfR * dfR + dfI * dfI))/2.);;
```

Comment:
for the complex case, I've had feedback from @archer that it would be better to compute the RMS complex value whose real part would be the RMS of the real part and imaginary part the RMS of the imaginar part. The resulting value RMS = RMS(Real) + i * RMS(Imag) would be such that abs(RMS) =RMS(abs(X)) because RMS(abs(X))^2 = mean(abs(X)^2) = mean(abs(Real + i * Imag)^2) = mean(Real^2 + Imag^2) = mean(Real^2) + mean(Imag^2) = RMS(Real)^2 + RMS(Imag)^2

Reply:
I confirm that from a user perspective the amplitude and phase must be decoupled.
For the amplitude, the need is on RMS(abs(X)) and the RMS(X) is useless.
For the phase,  the proposed formula will allow to compute Phase(RMS) as the phase of RMS(Real) + i * RMS(Phase). I think that this makes less sense than Phase(AVG), but user can already compute this with the standard kernel.

Reply:
Done